src/main/java/com/teragrep/rlp_03/context/frame/access/Lease.java

```diff
@@ -1,28 +1,29 @@
 package com.teragrep.rlp_03.context.frame.access;
 
+import java.util.concurrent.Phaser;
+
 public class Lease implements AutoCloseable {
 
     private final Access access;
-
-    private volatile boolean isOpen;
+    private final Phaser subPhaser;
     Lease(Access access) {
         this.access = access;
-        this.isOpen = true;
+        this.subPhaser = new Phaser(1);
```

Comment:
these should be of the Access's phaser, so when sub-phaser terminates, so terminates the one in Access

Reply:
will change to access pr